Mathlib/Data/List/Basic.lean

```diff
@@ -920,18 +920,19 @@ def reverseRecOn {motive : List α → Sort*} (l : List α) (nil : motive [])
   | [] => cast (congr_arg motive <| by simpa using congr(reverse $h.symm)) <|
       nil
   | head :: tail =>
-    have : tail.length < l.length := by
-      rw [← length_reverse l, h, length_cons]
-      simp [Nat.lt_succ]
     cast (congr_arg motive <| by simpa using congr(reverse $h.symm)) <|
       append_singleton _ head <| reverseRecOn (reverse tail) nil append_singleton
 termination_by l.length
+decreasing_by
+  simp_wf
+  rw [← length_reverse l, h, length_cons]
+  simp [Nat.lt_succ]
 #align list.reverse_rec_on List.reverseRecOn
 
 @[simp]
 theorem reverseRecOn_nil {motive : List α → Sort*} (nil : motive [])
     (append_singleton : ∀ (l : List α) (a : α), motive l → motive (l ++ [a])) :
-    reverseRecOn [] nil append_singleton = nil := rfl
+    reverseRecOn [] nil append_singleton = nil := reverseRecOn.eq_1 ..
```

Comment:
Mathlib typically tries to avoid explicitly mentioning these auto-generated equation lemmas. Is it possible to prove this using some unfolding/simping?

(If not, I don't think it's a deal breaker for this PR :smile:)

Reply:
Now that these are now longer internal (`.eq_1` instead of `._eq_1`) I wonder if that policy still should be there. My impression is that they have become “normal to use”.

This can be proven using other tactics, of course, or `unseal` + `rfl`. On the train though, and the mathlib cache is too much for the WiFi here :-)

Reply:
```suggestion
    reverseRecOn [] nil append_singleton = nil := by unfold reverseRecOn; rfl
```